settings.gradle

```diff
@@ -1,3 +1,12 @@
+import org.gradle.api.initialization.resolve.RepositoriesMode
+
+if (file("secrets.gradle").exists()) {
+	apply from: "secrets.gradle"
+} else {
+	ext.githubUsername = System.getenv("GITHUB_ACTOR")
+	ext.githubToken = System.getenv("GITHUB_TOKEN")
+}
```

Comment:
I'm wondering if rather than managing a separate `secrets.gradle` file, we just get devs to export these values to their envs in their zshrc/bashrc files? Then there's just one way to resolve these values regarless of the environment it's running in.

Reply:
Hmm, I'm not sure... Especially if it's just for this repo, I like keeping it contained. Any project setup that involves system-wide configuration feels like a bit of a red flag!

If you prefer to configure as environment variables, you can do - just don't create the `secrets.gradle` file. (I could put this in the readme explicitly)

Reply:
Yeah, I see your point. There's just something about secrets being written to a file in a project for each dev like this that makes me a little uneasy. I guess as long as the PAT has the correct scope, read only, it should be ok.